libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -2563,6 +2563,9 @@ std::string YulUtilFunctions::nextArrayElementFunction(ArrayType const& _type)
 			templ("advance", toCompactHexWithPrefix(size));
 			break;
 		}
+		case DataLocation::Transient:
+			solUnimplementedAssert(false, "Transient data location is only supported for value types.");
```

Comment:
By the way, I just noticed that we also have `solUnimplemented()`, which is much better because it's both shorter and does not suggest this is an assert:
```suggestion
			solUnimplemented("Transient data location is only supported for value types.");
```